src/app/dynamic/enterprise/quotas/component.ts

```diff
@@ -100,11 +103,41 @@ export class QuotasComponent implements OnInit {
     this.dataSource.filter = query;
   }
 
-  async add(): Promise<void> {
-    await lastValueFrom(
+  add(): void {
+    this._openDialog(null);
+  }
+
+  edit(quota: QuotaDetails): void {
+    this._openDialog(quota);
+  }
+
+  async delete({name}: QuotaDetails): Promise<void> {
+    const isConfirmed = await lastValueFrom(
+      this._matDialog
+        .open<ConfirmationDialogComponent, ConfirmationDialogConfig, boolean>(ConfirmationDialogComponent, {
+          data: {
+            title: 'Delete Quota',
+            message: `Delete quota for ${name}?`,
+            confirmLabel: 'Delete',
+          },
+        })
+        .afterClosed()
+    );
+
+    if (!isConfirmed) return;
+
+    await lastValueFrom(this._quotaService.deleteQuota(name));
+
+    this._notificationService.success(`Deleting the ${name} datacenter`);
+    this._quotaService.refreshQuotas();
+  }
+
+  private async _openDialog<T = void>(data: QuotaDetails): Promise<T> {
+    return lastValueFrom(
```

Comment:
Remove unnecessary async/await usages otherwise the return value always be promise due to `async` keyword and where this method is invoked expects promise to be returned.

```suggestion
  this._matDialog
      .open<ProjectQuotaDialogComponent, QuotaDetails, T>(ProjectQuotaDialogComponent, {
        panelClass: 'km-quota-dialog',
        data,
      })
      .afterClosed()
      .pipe(take(1))
      .subscribe(_ => {});
```